File: app/src/main/java/de/markhaehnel/rbtv/rocketbeanstv/utils/NetworkHelper.java
package de.markhaehnel.rbtv.rocketbeanstv.utils;

import java.io.IOException;
import java.net.InetAddress;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;

public final class NetworkHelper {
    public static boolean hasInternet() {
        try {
            InetAddress inetAddress = InetAddress.getByName("google.com");
            return !inetAddress.equals("");
        } catch (Exception e) {
            return false;
        }
    }
    public static String getContentFromUrl(String url) throws IOException {
        OkHttpClient client = new OkHttpClient();
        Request request = new Request.Builder().url(url).build();
        Response response = client.newCall(request).execute();

        return response.body().string();
    }
}

File: app/src/main/java/de/markhaehnel/rbtv/rocketbeanstv/utils/NetworkHelper.kt
package de.markhaehnel.rbtv.rocketbeanstv.utils

import java.io.IOException
import okhttp3.OkHttpClient
import okhttp3.Request

object NetworkHelper {
    fun hasInternet(): Boolean {
        try {
            getContentFromUrl("https://google.com")
            return true
        } catch (e: Exception) {
            return false
        }

    }

    @Throws(IOException::class)
    fun getContentFromUrl(url: String): String {
        val client = OkHttpClient()
        val request = Request.Builder().url(url).build()
        val response = client.newCall(request).execute()

        return response.body()!!.string()
    }
}
